### ISSUE TITLE
The directory on status line never changes

### ISSUE THREAD

**Issue by Irving-cl** [powerline/powerline#2109]
I installed `powerline-status` using pip on linux. However, when I use it, the directory never changes when I `cd` to another directory. If I enter basn again, it would change. But then it keeps the same again even I `cd` to another directory:
![Screenshot from 2020-07-01 11-44-37](https://user-images.githubusercontent.com/12945188/86200841-61e05d00-bb90-11ea-96d8-5a85450df7c4.png)

Why is that?

**Comment by PH111P** [comment]
That's strange. Could you please share some details of your setup/configuration?

**Comment by Irving-cl** [comment]
@PH111P 
I installed it with pip: `pip install --user powerline-setup`. And the python version is 2.7.

It's on a debian machine.

And I added the below lines to `~/.bashrc`:
```
powerline-daemon -q
POWERLINE_BASH_CONTINUATION=1
POWERLINE_BASH_SELECT=1
. {repository_root}/powerline/bindings/bash/powerline.sh
```
Of course I changed `{repository_root}` to the right position. Is there anything else I need to do?

Thanks in advance!

**Comment by PH111P** [comment]
This looks good. 
Do you have anything else in your config that overrides/changes `$PROMPT_COMMAND`? That might cause problems, see  https://powerline.readthedocs.io/en/latest/troubleshooting.html#bash-prompt-stopped-updating for some details.

**Comment by Irving-cl** [comment]
I sourced `powerline.sh` at the last in `~/.bashrc`. But I don't know if there's anything that would overrides `$PROPT_COMMAND` somewhere. I read the document:

> Make sure that powerline commands appear in $PROMPT_COMMAND: some users of $PROMPT_COMMAND have a habit of overwriting the value instead of prepending/appending to it. All powerline commands start with _powerline or powerline, e.g. _powerline_set_prompt.

Does it mean when I echo this variable, things like `_powerline_set_prompt` should appear? However when I echo the variable it doesn't:
![Screenshot from 2020-07-02 09-12-54](https://user-images.githubusercontent.com/12945188/86305459-4171da80-bc44-11ea-89ff-aa7e4368f4df.png)
Is this the reason?

And the command `powerline shell left` doesn't report anything, however it draws a correct status line like this(I have 2 status line now, and then goes back):
![Screenshot from 2020-07-02 09-18-56](https://user-images.githubusercontent.com/12945188/86305796-25226d80-bc45-11ea-9bb4-86a26dbab997.png)

**Comment by PH111P** [comment]
Yes indeed, it looks like that is the issue. For reference, the expected value of `$PROMPT_COMMAND` should be something like
```bash
> echo $PROMPT_COMMAND 
_powerline_status_wrapper _powerline_set_prompt
```

For debug purposes, you can try to source `{repository_root}/powerline/bindings/bash/powerline.sh` directly in bash with any `bashrc` disabled (`--norc`); if something really overrides `$PROMPT_COMMAND` in your bashrc, this should produce a working powerline.

**Comment by PH111P** [comment]
A quick search for the `__bp_interactive_mode` function turned up that this probably belongs to the script `bash-preexec`, which indeed overrides `$PROMPT_COMMAND` (and is thus incompatible with powerline).

**Comment by Irving-cl** [comment]
Thanks. It's indeed caused by this problem and it works if I source `powerline.sh` in a bash with `bashrc` disabled.

However, when I used the workaround: https://github.com/powerline/powerline/issues/1844#issuecomment-351828409 for this issue, I still some problems when I started a bash:
```
usage: powerline-daemon [-h] [-r MODULE] [-w WIDTH] [--last-exit-code INT]
                        [--last-pipe-status LIST] [--jobnum INT]
                        [-c KEY.KEY=VALUE] [-t THEME.KEY.KEY=VALUE]
                        [-R KEY=VAL] [-p PATH] [--socket ADDRESS]
                        ext [{left,right,above,aboveleft}]
```
It seems that this workaround doesn't work for me.

Besides I really cannot find where `bash-preexec` is.

Is there any solution for this problem?

**Comment by PH111P** [comment]
This is a bit hard to debug with the rather little information you provided about the contents of your `bashrc`. Regarding the workaround, the message you are seeing indicates that `powerline-daemon` (or `powerline`) is called incorrectly.

What you can try, is to add both missing functions in your workaround:
``` bash
precmd_functions+=(_powerline_status_wrapper _powerline_set_prompt)
```

And of course, there is always the (perhaps very tedious) method of commenting out every single line of your `bashrc` “one-by-one” to see where the error (the wrong value of $PROMPT_COMMAND) originates from. Also note that perhaps `/etc/bash.bashrc` may contain the root of your problems as well.
